TTS/tts/models/vits.py

```diff
@@ -1149,14 +1162,23 @@ def voice_conversion(self, y, y_lengths, speaker_cond_src, speaker_cond_tgt):
         assert self.num_speakers > 0, "num_speakers have to be larger than 0."
         # speaker embedding
         if self.args.use_speaker_embedding and not self.args.use_d_vector_file:
-            g_src = self.emb_g(speaker_cond_src).unsqueeze(-1)
-            g_tgt = self.emb_g(speaker_cond_tgt).unsqueeze(-1)
+            # <--- VOICE CONVERSION HOT FIX START --> #
+            if self.reference_speaker_id_exist:
+                g_src = self.emb_g(speaker_cond_src).unsqueeze(-1).unsqueeze(0)
+            elif self.reference_speaker_d_vector_exist:
```

Comment:
A model can use just self.args.use_speaker_embedding or self.args.use_d_vector_file we cant have two speaker embeddings approach in the same model. For this reason, this elif is useless. Considering that, "if self.args.use_speaker_embedding and not self.args.use_d_vector_file:" need to deal with all intern embedding code and "elif not self.args.use_speaker_embedding and self.args.use_d_vector_file:" with all external embedding (d_vector).

Given that a model can't have internal and external speakers embeddings could you elaborate on why we need to deal with d_vectors (F.normalize(speaker_cond_src).unsqueeze(-1)) inside of a conditional that only models trained with internal speakers embeddings will enter ("if self.args.use_speaker_embedding and not self.args.use_d_vector_file:" -- means that the model uses internal speaker embeddings and not uses external speakers embeddings)?

Reply:
Reading the issue #1672  I figured out that the user was trying to do voice conversion in a model trained with internal speaker embeddings without providing the --reference_speaker_idx parameter, then the inference code try to extract the speaker embedding using the speaker encoder. I think that we can transform this PR into just a raise error in the synthesis functions when the user provides the --reference_wav and uses internal speaker embedding (use_speaker_embedding=True) and do not provide the --reference_speaker_idx.

Reply:
Yes, I was thinking about similar stuff while making this PR. This is more like a bandaid for non-technical people who try to mess around.

Reply:
@p0p4k  Do you agree that a "raise" statement forcing the user to provide the --reference_speaker_idx would be enough?

Reply:
Yes, because then we are following the intended mechanism of this model. Either use internal embeddings when present or use an external speaker encoder.